v2/thirdpartyemailpassword/migration/bulk-user-import.mdx

```diff
@@ -0,0 +1,253 @@
+
+  The example password hash is in the following format `$f_scrypt$<passwordHash>$<salt>$m=<mem_cost>$r=<rounds>$s=<base64_salt_separator>`
+
+#### Step 4: Use the formatted password hash in the import JSON file
+
+```json
+{
+  "users": [
+    {
+      "loginMethods": [
+        {
+          "recipeId": "emailpassword",
+          "tenantIds": ["public"],
+          "isVerified": true,
+          "isPrimary": true,
+          "email": "johndoe@gmail.com",
+          "passwordHash": "$f_scrypt$9Y8ICWcqbzmI42DxV1jpyEjbrJPG8EQ6nI6oC32JYz+/dd7aEjI/R7jG9P5kYh8v9gyqFKaXMDzMg7eLCypbOA==$/cj0jC1br5o4+w==$m=14$r=8$s=Bw==",
+          "hashingAlgorithm": "firebase_scrypt",
+          "timeJoinedInMSSinceEpoch": 1713260578868
+        },
+        {
+          "recipeId": "thirdparty",
+          "tenantIds": ["public"],
+          "isVerified": true,
+          "isPrimary": false,
+          "email": "johndoe@gmail.com",
+          "thirdPartyId": "google",
+          "thirdPartyUserId": "106347997792363870000",
+          "timeJoinedInMSSinceEpoch": 1713260578868
+        }
+      ]
+    }
+  ]
+}
+```
+</details>
+
+## Step 2: Add the users for import
+
+To import users, make a POST request to the bulk import API with contents of the JSON file as the request body. The API supports up to 10,000 users per request. For larger files, submit multiple requests or use our [script](https://github.com/supertokens/supertokens-core/tree/master/bulkimport_scripts/add_users_for_bulk_import) to automate the process.
+
+A successful request will return a 200 response. If there are validation errors, you'll receive a 400 response with error messages, including the user's index in the input JSON. No users will be imported if validation errors occur. Retry after fixing the errors.
+
+<CoreInjector showTenantId={false} defaultValue="http://localhost:3567">
+
+```bash
+curl --location --request POST '^{coreInjector_uri_without_quotes}/bulk-import/users' \
+--header 'api-key: ^{coreInjector_api_key_without_quotes}' \
+--header 'Content-Type: application/json; charset=utf-8' \
+--data-raw '{
+  "users": [
+    {
+      "loginMethods": [
+        {
+          "recipeId": "emailpassword",
+          "tenantIds": ["public"],
+          "isVerified": true,
+          "isPrimary": true,
+          "email": "johndoe@gmail.com",
+          "passwordHash": "$argon2d$v=19$m=12,t=3,p=1$aGI4enNvMmd0Zm0wMDAwMA$r6p7qbr6HD+8CD7sBi4HVw",
+          "hashingAlgorithm": "argon2",
+          "timeJoinedInMSSinceEpoch": 1713260578868
+        },
+        {
+          "recipeId": "thirdparty",
+          "tenantIds": ["public"],
+          "isVerified": true,
+          "isPrimary": false,
+          "email": "johndoe@gmail.com",
+          "thirdPartyId": "google",
+          "thirdPartyUserId": "106347997792363870000",
+          "timeJoinedInMSSinceEpoch": 1713260578868
+        }
+      ]
+    }
+  ]
+}'
```

Comment:
have you tested this command? seems like the password hash contain $ might try to resolve to an environment variable. Please ensure this works fine.

Reply:
Yeah, it works. 

<img width="1073" alt="image" src="https://github.com/supertokens/docs/assets/22813027/5149d672-a481-4f2c-be06-a42e6a94987c">